akka-remote/src/main/scala/akka/remote/serialization/MiscMessageSerializer.scala

```diff
@@ -387,6 +425,36 @@ class MiscMessageSerializer(val system: ExtendedActorSystem) extends SerializerW
   private def deserializeStatusFailure(bytes: Array[Byte]): Status.Failure =
     Status.Failure(payloadSupport.deserializePayload(ContainerFormats.Payload.parseFrom(bytes)).asInstanceOf[Throwable])
 
+  private def deserializeAddressData(bytes: Array[Byte]): Address =
+    addressFromDataProto(WireFormats.AddressData.parseFrom(bytes))
```

Comment:
since we need to keep support for the existing AddressData, but also add support for the "top level serialized one" which we do in the new format